### ISSUE TITLE
inappmessaging not working

### ISSUE THREAD

**Issue by dodomui** [dpa99c/cordova-plugin-firebasex#347]
# Bug report

**Current behavior:**

Inappmessaging totally not working at both Android and iOS.

**Expected behavior:**

Expected to received inappmessage popup after I create a new campaign at firebase console.
Do I need any configuration to use inappmessaging?
![Screenshot 2020-03-23 at 5 38 30 PM](https://user-images.githubusercontent.com/19356577/77302973-22061000-6d2d-11ea-9be5-b37f6f65e458.png)

**Steps to reproduce:**

**Screenshots**

**Environment information**

- Cordova CLI version 
	- 9.0.0 (cordova-lib@9.0.1)
- Cordova platform version
	- Installed platforms:
  android 8.1.0
  ios 5.1.1
Available platforms: 
  browser ^6.0.0
  electron ^1.0.0
  osx ^5.0.0
  windows ^7.0.0
- Plugins & versions installed in project (including this plugin)
    - cordova-android-firebase-gradle-release 4.0.0 "cordova-android-firebase-gradle-release"
cordova-android-play-services-gradle-release 4.0.0 "cordova-android-play-services-gradle-release"
cordova-plugin-add-swift-support 2.0.2 "AddSwiftSupport"
cordova-plugin-androidx 1.0.2 "cordova-plugin-androidx"
cordova-plugin-androidx-adapter 1.1.0 "cordova-plugin-androidx-adapter"
cordova-plugin-browsertab 0.2.0 "cordova-plugin-browsertab"
cordova-plugin-buildinfo 4.0.0 "BuildInfo"
cordova-plugin-camera 4.1.0 "Camera"
cordova-plugin-compat 1.2.0 "Compat"
cordova-plugin-crop 0.3.1 "CropPlugin"
cordova-plugin-customurlscheme 4.4.0 "Custom URL scheme"
cordova-plugin-device 2.0.3 "Device"
cordova-plugin-facebook4 6.4.0 "Facebook Connect"
cordova-plugin-file 6.0.2 "File"
cordova-plugin-firebasex 9.0.1-cli "Google Firebase Plugin"
cordova-plugin-geolocation 4.0.2 "Geolocation"
cordova-plugin-googleplus 8.4.0 "Google SignIn"
cordova-plugin-inappbrowser 3.1.0 "InAppBrowser"
cordova-plugin-ionic-keyboard 2.2.0 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 4.1.3 "cordova-plugin-ionic-webview"
cordova-plugin-nativestorage 2.3.2 "NativeStorage"
cordova-plugin-qrscanner 3.0.1 "QRScanner"
cordova-plugin-splashscreen 5.0.3 "Splashscreen"
cordova-plugin-statusbar 2.4.3 "StatusBar"
cordova-plugin-whitelist 1.3.4 "Whitelist"
cordova-sqlite-storage 4.0.0 "Cordova sqlite storage plugin - cordova-sqlite-storage plugin version"
cordova-support-android-plugin 1.0.1 "cordova-support-android-plugin"
cordova-support-google-services 1.4.0 "cordova-support-google-services"
- Dev machine OS and version, e.g.
    - OSX
        - `Catalina 10.15.3`
        
_Runtime issue_
- Device details
    - _e.g. iPhone X, Samsung Galaxy S8, iPhone X Simulator, Pixel XL Emulator_
- OS details
    - _e.g. iOS 12.2, Android 9.0_	
	
- Node JS version
    - `v12.16.1`

**Related code:**
```
insert any relevant code here such as plugin API calls / input parameters
```

**Console output**
<details>
<summary>console output</summary>

```

// Paste any relevant JS/native console output here

```

</details><br/><br/>

**Other information:**

**Comment by dpa99c** [comment]
> cordova-plugin-firebasex 9.0.1-cli "Google Firebase Plugin"

You are using the CLI build version which does not contain the Inapp Messaging component due to a Cordova iOS bug which causes the build to fail when this component is included.

TL;DR: if you want to use Inapp Messaging, build using `cordova-plugin-firebasex@9.0.1` using Xcode

**Comment by dodomui** [comment]
Hi, I tried to installed cordova-plugin-firebasex@9.0.1, but still unable to received the inappmessaging.
cordova-plugin-firebasex 9.0.1 "Google Firebase Plugin"

Also, receive following error while run ionic cordova build ios

`error: Resource "/Users/../Library/Developer/Xcode/DerivedData/edhpzwhnohkxypeinxxprdjmgvfn/Build/Products/Debug-iphonesimulator/FirebaseInAppMessaging/InAppMessagingDisplayResources.bundle" not found. Run 'pod install' to update the copy resources script.`

**Comment by dpa99c** [comment]
> `ionic cordova build ios`

You are trying to build using the command line hence you are getting this error - `cordova-plugin-firebasex@9.0.1` does not work with CLI builds for the reason I have outline above .

You must build projects containing `cordova-plugin-firebasex@9.0.1` using the Xcode IDE (open `YourProject.xcworkspace in Xcode).

`cordova-plugin-firebasex@9.0.1` has been regression tested using the [example project](https://github.com/dpa99c/cordova-plugin-firebasex-test) and Inapp Messaging works fine on both the Android & iOS platforms.

**Comment by dodomui** [comment]
Thanks for your fast respond.
Sorry, I'm new to Ionic, may I know how to build by Xcode directly?

After I updated my ionic code, do I need to run CLI builds first?
Or after amend my ionic code, just open Xcode -> Product -> Build?

Thanks a lot if you could kindly reply on this.

**Comment by dpa99c** [comment]
You can't do an iOS build at all using the CLI - it will always fail due to an underlying issue with Cordova when including the Firebase Inapp Messaging SDK component via Cocoapods.

After making code changes, you will have to prepare your build assets using the CLI: `ionic cordova prepare ios`
And build/run in Xcode: `Xcode -> Product ->Build/Run`

**Comment by dodomui** [comment]
Thanks a lot. 
It's working on iOS now.

**Comment by dodomui** [comment]
Do I need same steps for Android? Still unable to receive inappmessaging in android.

**Comment by dpa99c** [comment]
Android builds fine from the CLI. Make sure you have the removed/re-added the Android platform (`cordova platform rm android & cordova platform add android`) to ensure the change from `9.0.1-cli` to `9.0.1` has been applied.

You can confirm the that the Inapp Messaging component [included by this plugin](https://github.com/dpa99c/cordova-plugin-firebasex/blob/master/plugin.xml#L84) is present in `platforms/android/app/build.gradle`.

**Comment by dodomui** [comment]
I did try rm and re-add cordova android and inappmessaging component is present.

But In App Messaging still doesn't work on android, only iOS is working fine.
Will I miss anything?

```
implementation(project(path: ":CordovaLib"))
    implementation "androidx.browser:browser:1.0.0"
    implementation "androidx.legacy:legacy-support-v4:1.0.0"
    implementation "com.soundcloud.android:android-crop:1.0.0@aar"
    implementation "com.facebook.android:facebook-android-sdk:5.5.2"
    implementation "com.google.android.gms:play-services-tagmanager:17.0.0"
    implementation "com.google.android.gms:play-services-auth:17.0.0"
    implementation "com.google.firebase:firebase-analytics:17.2.1"
    implementation "com.google.firebase:firebase-messaging:20.0.0"
    implementation "com.google.firebase:firebase-config:19.0.3"
    implementation "com.google.firebase:firebase-perf:19.0.1"
    implementation "com.google.firebase:firebase-auth:19.1.0"
    implementation "com.google.firebase:firebase-inappmessaging-display:19.0.3"
    implementation "com.google.firebase:firebase-firestore:21.4.0"
    implementation "com.crashlytics.sdk.android:crashlytics:2.10.1"
    implementation "com.crashlytics.sdk.android:crashlytics-ndk:2.1.1"
    implementation "com.google.code.gson:gson:2.8.6"
    implementation "com.google.android.gms:play-services-auth:15.0.1"
    implementation "com.google.android.gms:play-services-identity:15.0.1"
    implementation "androidx.annotation:annotation:1.0.0"
```

**Comment by dpa99c** [comment]
> implementation "com.google.firebase:firebase-inappmessaging-display:19.0.3"

That shows the SDK component has been included in the build. There is no API for Inapp Messaging, so it should just work.

Bear in mind when testing it that messages are by default only fetched from the server once per day. I
f you are using the device's app instance ID to send a test message to a specific test device, you may still need to background-then-foreground or exit-the-restart the app before the message is displayed.

**Comment by dodomui** [comment]
Yeah, I did try restart and re-enter the app. 
Understand the message show once per day only therefore I created multiple campaign for each individual testing.

Just no luck, only iOS is working but not android.

Seem I'm the only one facing this issue on android, let me further check on this hope I got luck on it.
Thanks @dpa99c for your kindly help.

Stay healthy from the virus ^_^

**Comment by dodomui** [comment]
UPDATED: inappmessage is working in both Android and iOS.
Thanks and sorry for your times.
